### PR TITLE
refactor: optimize clean up script for outputs and node_modules

### LINKED ISSUE #1091: Implement a cleanup script
### Context
As a developer, I often find myself needing to clean up the project directory to ensure consistency, such as removing unnecessary `/dist` or `/out` directories and `node_modules` directories. Currently, this cleanup process is manual and cumbersome, requiring developers to delete these directories individually.

### To Do
Implement a script that automates the cleanup process, preferably utilizing NX tools such as generators or executors. A `clean` script already exists in the project, but it needs improvement to effectively handle the cleanup tasks. This existing script can serve as a starting point for the implementation.

### Acceptance Criteria
- [ ] Script is implemented to automate the cleanup process of the project directory.
- [ ] The script effectively removes all `/dist` or `/out` directories and `node_modules` directories within the project.
- [ ] Utilize NX tools, such as generators or executors, where applicable to streamline the implementation process.
- [ ] The script is tested and verified to work reliably across different environments and project configurations.
- [ ] NX reset cache

### PR DESCRIPTION
**Pull request by BenPag**
Closes #1091 

## Proposed Changes

- add shared target "clean" in `nx.json`
- hide `scripts` in `package.json` from nx

### CODE REVIEW

**Review by janivo — CHANGES_REQUESTED** [review]
I noticed two things when running the scripts:
- `packages/elements/.stencil` & `.nx` still exist
- the `clean` script seems not to work quite right, I get the error below

```
$ nx run-many --target=clean && nx reset && shx rm -rf node_modules

  ✔  nx run elements-angular:clean (816ms)
  ✔  nx run elements-react:clean (821ms)
  ✔  nx run elements:clean (832ms)







  ✔  nx run elements-vue:clean (698ms)
  ✔  nx run ui-patterns:clean (703ms)
  ✔  nx run elements-angular-example:clean (695ms)






  ✔  nx run elements-react-example:clean (693ms)

  ✔  nx run elements-vue-example:clean (708ms)


  ✔  nx run ui-patterns-playground:clean (968ms)


————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
  ✖  nx run storybook:clean
     NX   Daemon process terminated and closed the connection
    
    Please rerun the command, which will restart the daemon.
    If you get this error again, check for any errors in the daemon process logs found in: /Users/arbeit/Projects/elements/.nx/cache/d/daemon.log
    
    Warning: command "nx run storybook:clean:dist" exited with non-zero status code
    





  ✔  nx run landingpage:clean (905ms)
  ✔  nx run workspace:clean (723ms)

————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 NX   Ran target clean for 12 projects (3s)

  ✔    11/12 succeeded [0 read from cache]

  ✖    1/12 targets failed, including the following:
    - nx run storybook:clean

error Command failed with exit code 1.
```

File: package.json [hunk 1]
```diff
@@ -10,16 +10,20 @@
...
+    "lint": "nx affected --target=lint --output-style=stream",
+    "lint:fix": "nx affected --target=lint --configuration=fix --output-style=stream",
+    "format": "nx format:check",
+    "format:fix": "nx format:write",
     "qc": "yarn lint && yarn format && yarn build && yarn test",
-    "clean": "nx run-many --target=clean --exclude workspace && shx rm -rf dist",
-    "clean:node-modules": "nx run-many --target=clean --configuration=nodejs --exclude workspace && shx rm -rf node_modules"
+    "clean": "nx run-many --target=clean && nx reset && shx rm -rf node_modules",
+    "clean:dist": "nx reset && nx run-many --target=clean --configuration=dist",
+    "clean:node": "nx run-many --target=clean --configuration=node && nx reset && shx rm -rf node_modules"
```
Comment (janivo) [inline package.json:23]: Is there a reason we execute `nx reset` in `clean:dist` before everything else? (in contrast to `clean` & `clean:node`)

Comment (BenPag) [reply package.json]: No, there is not. I changed this.

**Comment by BenPag** [comment]
> I noticed two things when running the scripts:
> 
> * `packages/elements/.stencil` & `.nx` still exist
> * the `clean` script seems not to work quite right, I get the error below
> 
> ```
> $ nx run-many --target=clean && nx reset && shx rm -rf node_modules
> 
>   ✔  nx run elements-angular:clean (816ms)
>   ✔  nx run elements-react:clean (821ms)
>   ✔  nx run elements:clean (832ms)
> 
> 
> 
> 
> 
> 
> 
>   ✔  nx run elements-vue:clean (698ms)
>   ✔  nx run ui-patterns:clean (703ms)
>   ✔  nx run elements-angular-example:clean (695ms)
> 
> 
> 
> 
> 
> 
>   ✔  nx run elements-react-example:clean (693ms)
> 
>   ✔  nx run elements-vue-example:clean (708ms)
> 
> 
>   ✔  nx run ui-patterns-playground:clean (968ms)
> 
> 
> ————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
>   ✖  nx run storybook:clean
>      NX   Daemon process terminated and closed the connection
>     
>     Please rerun the command, which will restart the daemon.
>     If you get this error again, check for any errors in the daemon process logs found in: /Users/arbeit/Projects/elements/.nx/cache/d/daemon.log
>     
>     Warning: command "nx run storybook:clean:dist" exited with non-zero status code
>     
> 
> 
> 
> 
> 
>   ✔  nx run landingpage:clean (905ms)
>   ✔  nx run workspace:clean (723ms)
> 
> ————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
> 
>  NX   Ran target clean for 12 projects (3s)
> 
>   ✔    11/12 succeeded [0 read from cache]
> 
>   ✖    1/12 targets failed, including the following:
>     - nx run storybook:clean
> 
> error Command failed with exit code 1.
> ```

I added the .stencil dir. Regarding your error I will investigate...

**Review by janivo — APPROVED** [review]

**Pull request merged**